ProductService/src/main/java/com/metaphorce/exam/ecommerce/product/controller/ProductController.java

```diff
@@ -0,0 +1,138 @@
+package com.metaphorce.exam.ecommerce.product.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.metaphorce.exam.ecommerce.product.dto.ProductDTO;
+import com.metaphorce.exam.ecommerce.product.mapper.ProductMapper;
+import com.metaphorce.exam.ecommerce.product.model.impl.Product;
+import com.metaphorce.exam.ecommerce.product.service.impl.ProductService;
+
+import jakarta.validation.Valid;
+import lombok.extern.slf4j.Slf4j;
+
+@RequestMapping("/product")
+@RestController
+@Slf4j
+public class ProductController {
+	@Autowired
+	private ProductService service;
+	
+	/**
+	 * Implements the HTTP Post to create a Product
+	 * @param Product Data to insert
+	 * @return New Product create
+	 * @author Juan Chavez
+	 * @since Jun/11/2024 
+	 */
+	@PostMapping
+	public ResponseEntity<ProductDTO> createProduct(@Valid @RequestBody ProductDTO productRequest) {
+		log.info("Product to create: " + productRequest);
+		ResponseEntity<ProductDTO> response = null;
+		Product newProduct = this.service.create(ProductMapper.mapper.toEntity(productRequest));
+
+		if(newProduct != null) {
+			response = ResponseEntity.ok(ProductMapper.mapper.toDTO(newProduct));
+			log.info("Product created: " + newProduct);
+		} else {
+			response = ResponseEntity.badRequest().build();
+			log.info("Error to create Product: " + productRequest);
+		}
+
+		return response;
+	}
+
+	/**
+	 * Implements the HTTP Get to find all Products
+	 * @return List of Products
+	 * @author Juan Chavez
+	 * @since Jun/11/2024 
+	 */
+	@GetMapping
+	public ResponseEntity<List<ProductDTO>> getAllProducts() {
+		log.info("Get All Products ...");
+		ResponseEntity<List<ProductDTO>> response = null;
+		List<Product> listProducts = this.service.getAll();
+
```

Comment:
I don't understand why check if the value is null. It would be better to throw exceptions in the service when it doesn't find a certain resource; also, add a global exception handler.

In the case of not being able to create a resource, I think it's better to let a 500 status be generated because it probably means there's something wrong with the system.